libs/storage/Tsavorite/cs/src/core/Allocator/GenericAllocatorImpl.cs

```diff
@@ -433,13 +434,24 @@ private void WriteAsync<TContext>(long flushPage, ulong alignedDestinationAddres
                             if (ValueHasObjects() && !record.info.Tombstone)
                             {
                                 long pos = ms.Position;
-                                valueSerializer.Serialize(ref src[i].value);
-
-                                // Store the value address into the 'buffer' AddressInfo image as an offset into 'ms'.
-                                value_address->Address = pos;
-                                value_address->Size = (int)(ms.Position - pos);
-                                addr.Add((long)value_address);
-                                endPosition = pos + value_address->Size;
+                                try
+                                {
+                                    valueSerializer.Serialize(ref src[i].value);
```

Comment:
Try/catch around only this, if failed then reset pos and retry after setting src[i].value to default. No need to set invalid etc.